strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/repository/ProxyRepositoryProvider.java

```diff
@@ -136,29 +133,23 @@ public ArtifactInputStream getInputStream(String storageId,
             
             // TODO: Add a policy for validating the checksums of downloaded artifacts
             // TODO: Validate the local checksum against the remote's checksums
-
             fileSystemProvider.restoreFromTemp(artifactPath);
 
             // 1 b) If it exists on the remote, serve the downloaded artifact
-
-            ArtifactInputStream ais = new ArtifactInputStream(null, Files.newInputStream(artifactPath));
-            ais.setLength(total);
-
-            return ais;
+            return new ArtifactInputStream(null, Files.newInputStream(artifactPath));
         }
     }
 
     @Override
-    public OutputStream getOutputStream(String storageId,
+    public ArtifactOutputStream getOutputStream(String storageId,
                                         String repositoryId,
                                         String artifactPath)
             throws IOException
```

Comment:
Could you please re-indent all the parameters across your pull which aren't properly indented? Thanks!